### ISSUE TITLE
[FEATURE] Add support for PutComposableIndexTemplateRequest while resolving request

### ISSUE THREAD

**Issue by shikharj05** [opensearch-project/security#1863]
**Is your feature request related to a problem?**
Security plugin doesn't support [PutComposableIndexTemplateRequest](https://github.com/opensearch-project/OpenSearch/blob/main/client/rest-high-level/src/main/java/org/opensearch/client/indices/PutComposableIndexTemplateRequest.java) while [resolving](https://github.com/opensearch-project/security/blob/78837b610bc35de8c07b46b33b7f8a5332f732e7/src/main/java/org/opensearch/security/resolver/IndexResolverReplacer.java#L525) request/index permissions. This causes permissions to be evaluated against all(*) indices.
Hence, a user/role with permissions limited to specific index patterns sees 403s when an index template request is performed.

**What solution would you like?**
Support for specific index patterns permissions with index template actions 

**What alternatives have you considered?**
NA

**Do you have any additional context?**
NA

**Comment by peternied** [comment]
Thanks for filing, we would be happy to accept a pull request for this issue.

**Comment by DarshitChanpura** [comment]
@shikharj05 Can you list the steps to reproduce this issue?

**Comment by cliu123** [comment]
@DarshitChanpura The reproduce steps seem to be in [the description of another issue](https://github.com/opensearch-project/documentation-website/issues/619).

**Comment by shikharj05** [comment]
@DarshitChanpura - the issue can be reproduced using the steps here - https://github.com/opensearch-project/documentation-website/issues/619

You will see a debug log line -
```
class org.opensearch.action.admin.indices.template.put.PutIndexTemplateRequest 
not supported (It is likely not a indices related request)
```
However, since the workaround in above issue suggests adding **indices:admin/template/*** permission to cluster permissions, user can create index template for any index.

@cliu123 updated the description to include the right link.

**Comment by peternied** [comment]
[Triage] We've looked into this request for supporting this permissions on an index level, and with the current architecture this proposal would create many problems.  For example templates do not apply to existing indices that can be permissioned, instead they are associated with a patterns of future created indices, who can control reducing or increasing the scope of these patterns is a hard thing to describe with our current model that would lead to more confusing behavior.

As we aren't going to implement changes in this space, closing out this issue.


Note; We did find some bugs this space see https://github.com/opensearch-project/security/issues/1884, and will be addressing that issue shortly.
